Modules/Scripted/ShapeAnalysisModule/ShapeAnalysisModule.py

```diff
@@ -1315,21 +1313,21 @@ def setup(self):
 
     ## Mesh To Label Map: Transform model in label map
     cli_nodes = list() # list of the nodes used in the Mesh to Label Map step
-    cli_filepaths = list() # list of the node filepaths used in the Mesh to Label Map step
-    LabelMapDirectory = outputDirectory + "/LabelMap"
-    LabelMapOutputFilepath = LabelMapDirectory + "/" + self.inputFilename + ".nrrd"
+    cli_dirnames = list() # list of the directory pathes where the nodes used in the Mesh to Label Map step are stored
+    MeshToLabelMapOutputDirectory = outputDirectory + "/Step0_MeshToLabelMap"
+    MeshToLabelMapOutputFilename = self.inputRootname + ".nrrd"
+    MeshToLabelMapOutputFilepath = os.path.join(MeshToLabelMapOutputDirectory, MeshToLabelMapOutputFilename)
     if not self.skip_meshToLabelMap:
       # Setup of the parameters of the CLI
       self.ID += 1
 
       cli_parameters = {}
 
-      inputFilepath = inputDirectory + '/' + self.CaseInput
-      model_input_node = MRMLUtility.loadMRMLNode(inputFilepath, 'ModelFile')
+      model_input_node = MRMLUtility.loadMRMLNode(self.inputRootname, inputDirectory, self.CaseInput, 'ModelFile')
 
       cli_parameters["mesh"] = model_input_node
 
-      meshtolabelmap_output_node = MRMLUtility.addnewMRMLNode("output_MeshToLabelMap", slicer.vtkMRMLLabelMapVolumeNode())
+      meshtolabelmap_output_node = MRMLUtility.createNewMRMLNode(self.inputRootname, slicer.vtkMRMLLabelMapVolumeNode())
```

Comment:
I think the second parameter is a string just like: https://github.com/Kitware/SlicerSALT/blob/c988d2d722c2eeb9770bbe5c4418a776f8cdc69d/Modules/Scripted/ShapeAnalysisToolBox/DataImporter.py#L52

Reply:
I modified the function 'createNewMRMLNode': in order to add nodes to the MRML scene, I use now vtkMRMLScene::AddNode(vtkMRMLNode * nodeToAdd) instead of vtkMRMLScene::AddNewNodeByClass(std::string className). 
The change is to make SPHARM-PDM extension work with the Slicer release (4.6).

Reply:
I see.. But the [if-else statements](https://github.com/NIRALUser/SPHARM-PDM/pull/24/files#diff-999f279ddadd1031c967c0ccdc5be2f5R38) after the AddNode line assume that the 'mrl_type' is a string. And hence the concern. I would prefer using `slicer.mrmlScene.AddNewNodeByClass`, but I understand the concern for keeping compatibility with Slicer 4.6 . Not sure what the best approach is here. Maybe we could wait to do a transition to the new method when 4.7 becomes an official release. Till that time we could get the class name using `node.GetClassName()` method and use that in the if-else block.

Reply:
I amended the last commit to modify the function `createNewMRMLNode`. It now uses`node.GetClassName()` method in the if-else block as you suggested. :)